### PR TITLE
Use systemd and other updates on stable-2.332 branch

### PR DESCRIPTION
**Pull request by MarkEWaite**
## Use systemd and other updates on stable-2.332 branch

The stable-2.332 branch was already configured to deliver 2.332.1 with `systemd`.  This change includes the most recent updates that have been delivered in at least one weekly release.  It does not include updates from the master branch that have not yet been delivered in at least one weekly release.

See [Jenkins developers mailing list discussion](https://groups.google.com/g/jenkinsci-dev/c/mRxFPGfuljA/m/y7Rb96V5AgAJ) for more details.

## Changes included

* Test at least once a week
* Workaround broken centos stream 9 repo setup
* Support for `systemd` (#266)
* Support port numbers less than 1024 in `systemd` (#285)
* Regenerate test key

## Checklist

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Review by basil — APPROVED** [review]

File: Jenkinsfile
Comment (timja) [inline Jenkinsfile:4]: I don’t think we want this on non master branches

Comment (MarkEWaite) [reply Jenkinsfile]: Can you explain further why we would not want to run the test automation once a week on any branch that is active?

Once a branch becomes inactive (stable-2.319, stable-2.303, stable-2.289), we'd want to disable the weekly trigger of the job, but it seems safe and wise to assure that the test automation is working on both the weekly branch and the current stable branch.

I've submitted the revert of that change on the stable-2.332 branch as ea933557fa3d16d44a21e1153ccefc770126f592

**Review by timja — APPROVED** [review]

File: Jenkinsfile
Comment (timja) [reply Jenkinsfile]: I think it's a waste of CI and we will forget to disable it. :)

Should mainline an on master only cron trigger I think.

**Comment by basil** [comment]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Pull request merged**
